pkg/controllers/machine_controller.go

```diff
@@ -456,17 +456,21 @@ func (r *MachineReconciler) reconcileMachine(ctx context.Context, id string) err
 
 	log.V(1).Info("Reconciling domain")
 	state, volumeStates, nicStates, err := r.reconcileDomain(ctx, log, machine)
+
+	machine.Status.VolumeStatus = volumeStates
 	if err != nil {
+		if _, locErr := r.machines.Update(ctx, machine); locErr != nil {
```

Comment:
Does it still make sense to proceed with the update even if volumeStates is nil?

Reply:
i don't think volumeStatus can be nil. I tried fixed every return values for that case.

Reply:
https://github.com/ironcore-dev/libvirt-provider/blob/b7126900194d52d4d6debcd233f4d1134f520f8b/pkg/controllers/machine_controller.go#L487
Here, it can be nil.

Reply:
fixed

Reply:
it can be nil, even in below:
- https://github.com/ironcore-dev/libvirt-provider/blob/c5809e04a24f80fd1d449af792cc9334b2989e3c/pkg/controllers/machine_controller.go#L521
- https://github.com/ironcore-dev/libvirt-provider/blob/c5809e04a24f80fd1d449af792cc9334b2989e3c/pkg/controllers/machine_controller.go#L526
- https://github.com/ironcore-dev/libvirt-provider/blob/c5809e04a24f80fd1d449af792cc9334b2989e3c/pkg/controllers/machine_controller.go#L561
- https://github.com/ironcore-dev/libvirt-provider/blob/c5809e04a24f80fd1d449af792cc9334b2989e3c/pkg/controllers/machine_controller.go#L566
- https://github.com/ironcore-dev/libvirt-provider/blob/c5809e04a24f80fd1d449af792cc9334b2989e3c/pkg/controllers/machine_controller.go#L572

I suggest to update when the volumeStatus is not nil.

Reply:
Ok i will create function for it

Reply:
fixed